server/src/catalog.rs

```diff
@@ -65,147 +101,206 @@ impl Catalog {
+        let name = if let Some(index) = file_or_folder_name.find('.') {
+            file_or_folder_name[0..index].to_string()
+        } else {
+            file_or_folder_name
+        };
+
+        Self {
+            name,
+            path: file_or_folder_path,
         }
     }
+}
 
-    // Read time series metadata.
-    // TODO: read tids and gids so data from time series groups can be correctly decompressed.
-    let time_series_file = table_folder.clone() + "/time_series.parquet";
-    let path = Path::new(&time_series_file);
-    if let Ok(file) = File::open(&path) {
+// TODO: update after implementing the compression component.
+/// Metadata required to query a model table.
+#[derive(Debug)]
+pub struct ModelTableMetadata {
+    /// Name of the model table.
+    pub name: String,
+    /// Location of the model table's segment folder in an `ObjectStore`.
+    pub segment_path: ObjectStorePath,
+    // TODO: remove segment_file_legacy_schema when refactoring query engine.
+    /// Schema of the Apache Parquet files used by the [JVM-based version of
+    /// ModelarDB] for storing segments.
+    ///
+    /// [JVM-based version of ModelarDB]: https://github.com/modelardata/ModelarDB
+    pub segment_file_legacy_schema: SchemaRef,
+    /// Cache that maps from time series id to sampling interval for all time
+    /// series in the table.
+    pub sampling_intervals: Int32Array,
+    /// Cache that maps from time series id to members for all time series in
+    /// the table. Each `Array` in `denormalized_dimensions` is a level in a
+    /// dimension.
+    pub denormalized_dimensions: Vec<ArrayRef>,
+}
+
+impl ModelTableMetadata {
+    /// Read and check the metadata for a model table produced by the [JVM-based
+    /// version of ModelarDB]. Return `ParquetError` if the metadata cannot be
+    /// read or if the model table uses unsupported model types.
+    ///
+    /// [JVM-based version of ModelarDB]: https://github.com/modelardata/ModelarDB
+    fn try_new(
+        folder_name: String,
+        folder_path: String,
+        segment_file_legacy_schema: &SchemaRef,
+    ) -> Result<Arc<Self>, ParquetError> {
+        // Pre-compute the path to the segment folder in the object store.
+        let segment_path = Self::create_object_store_path_to_segment_folder(&folder_path)?;
+
+        // Ensure only supported model types are used.
+        Self::check_model_type_metdata(&folder_path)?;
+
+        // Read time series metadata.
+        let time_series_file = folder_path.clone() + "/time_series.parquet";
+        let path = Path::new(&time_series_file);
+        if let Ok(file) = File::open(&path) {
+            let rows = Self::read_entire_parquet_file(&folder_name, file)?;
+            let sampling_intervals = Self::extract_and_shift_int32_array(&rows, 2)?;
+            let denormalized_dimensions =
+                Self::extract_and_shift_denormalized_dimensions(&rows, 4)?;
```

Comment:
I am not sure that the "REDD_MDB_Parquet" data even has this "denormalized_dimensions" column since the time series file only has 4 columns. What is this column used for?

Reply:
Added the following comment next to `denormalized_dimensions` "// Columns with metadata as strings." In the JVM-based version of ModelarDB the dimensions (e.g., `type: Utf8, entity: Utf8, category: Utf8, concrete: Utf8` in a table like `tid: Int32, timestamp: Timestamp(Millisecond, None), value: Float32, type: Utf8, entity: Utf8, category: Utf8, concrete: Utf8`) was also used for statically grouping time series so they could be compressed together as described in the ICDE 2021 paper ([IEEE](https://ieeexplore.ieee.org/document/9458830), [arXiv (preprint)](https://arxiv.org/abs/1903.10269)). However, static grouping is not implemented as it should be superseded by the planned model combiner.